python/ray/tune/execution/trial_runner.py

```diff
@@ -197,24 +227,42 @@ def _serialize_and_write():
         else:
             exclude = ["*/checkpoint_*"]
 
+        synced = False
         if self._syncer:
             # Todo: Implement sync_timeout for experiment-level syncing
             # (it is currently only used for trainable-to-cloud syncing)
             if force:
                 # Wait until previous sync command finished
                 self._syncer.wait()
-                self._syncer.sync_up(
+                synced = self._syncer.sync_up(
                     local_dir=self._local_dir,
                     remote_dir=self._remote_dir,
                     exclude=exclude,
                 )
             else:
-                self._syncer.sync_up_if_needed(
+                synced = self._syncer.sync_up_if_needed(
                     local_dir=self._local_dir,
                     remote_dir=self._remote_dir,
                     exclude=exclude,
                 )
 
+        if synced:
+            self._trial_num_checkpoints_since_last_sync.clear()
+            if now - self._last_sync_time < self._excessive_sync_threshold:
+                logger.warning(
+                    "Experiment checkpoint syncing has been triggered multiple "
+                    f"times in the last {self._excessive_sync_threshold} seconds. "
+                    "A sync will be triggered whenever a trial has checkpointed "
+                    "more than `num_to_keep` times since last sync or if "
+                    f"{self._syncer.sync_period} seconds have passed since last "
+                    "sync. If you have set `num_to_keep` in your `CheckpointConfig`, "
+                    "consider increasing the checkpoint frequency or keeping more "
+                    "checkpoints. You can supress this warning by changing the "
+                    "`TUNE_WARN_EXCESSIVE_EXPERIMENT_CHECKPOINT_SYNC_THRESHOLD_S` "
+                    "environment variable."
+                )
+            self._last_sync_time = now
```

Comment:
The sync could have taken a long time, so we should probably update the last sync time to a fresh `time.time()`

Reply:
(please add a comment for that for future reference)